### PR TITLE
SB-51: search bar menu fix

### PR DESCRIPTION
**Pull request by VicAv99**
## Feature/Problem

The accessibility was off on the search bar menu.

## Solution

added `event.stopImmediatePropagation()` throughout because since buttons are so close together, events were getting in the way. I also created a child component for the buttons.

## Areas of Impact

searchbar.component

## UI Images

image

## Manual Testing

- Verify CI passes

### CODE REVIEW

File: client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts [hunk 1]
```diff
@@ -37,23 +31,20 @@ export class SearchbarComponent implements OnDestroy, OnInit {
...
 
-  selectCustom(searchType: string, index = 0) {
+  selectCustom({ searchType, index }) {
     const datepickerOption = 2;
 
-    this.form.patchValue({ searchType });
-
     if (index === datepickerOption) {
       this.patchSelectedDate();
+
```
Comment (eccentricnode) [inline client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts]: Unnecessary whitespace...

File: client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts [hunk 2]
```diff
@@ -37,23 +31,20 @@ export class SearchbarComponent implements OnDestroy, OnInit {
...
+      return;
     }
+
```
Comment (eccentricnode) [inline client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts]: Also, unnecessary white-space

File: client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts (hunk 1 above)
Comment (VicAv99) [reply client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts]: We have a rule set in tslint where, whenever we use a `return` it should have a space on top of it.

File: client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts (hunk 2 above)
Comment (VicAv99) [reply client/libs/ui-libraries/src/lib/searchbar/searchbar.component.ts]: Usually, we add a space when after an if statement to show clarity that something may happen after it.

**Comment by VicAv99** [comment]
> Good PR... Needs some work tho

Where is the work needed?
